### ISSUE TITLE
[Fedora Rawhide] Mold fails to compile Mesa drivers with LTO

### ISSUE THREAD

**Issue by Nanotwerp** [rui314/mold#627]
Everything compiles perfectly until this error occurs:

`clang-14: error: linker command failed with exit code 1 (use -v to see invocation)`

This happens regardless of the ulimit value set. I'm thinking that maybe this has to do with how the version of libtbb on Fedora is outdated.

**Comment by rui314** [comment]
It's likely the same issue as https://github.com/rui314/mold/issues/600.

What if you build mold yourself? If you build mold yourself, it'll use the vendored, patched version of tbb.

**Comment by Nanotwerp** [comment]
> It's likely the same issue as #600.
> 
> What if you build mold yourself? If you build mold yourself, it'll use the vendored, patched version of tbb.

I just compiled mold using Fedora's spec file of it, and the issue still happened. The spec file keeps the default behavior of mold: bundling and linking against the bundled libtbb (relevant part from the spec file):

```
# API-incompatible with older tbb 2020.3 currently shipped by Fedora:
# https://bugzilla.redhat.com/show_bug.cgi?id=2036372
Provides:	bundled(tbb) = 2021.5
```
Also an example of output when it compiles:

`cd /home/x/mold/mold-1.4.0/out/tbb/src/tbb && /usr/bin/clang++ -D__TBB_BUILD -D__TBB_DYNAMIC_LOAD_ENABLED=0 -D__TBB_SOURCE_DIRECTLY_INCLUDED=1 -D__TBB_USE_ITT_NOTIFY -I/home/x/mold/mold-1.4.0/third-party/tbb/src/tbb/../../include -Wno-sign-compare -D__TBB_DYNAMIC_LOAD_ENABLED=0 -O2 -g -DNDEBUG -fPIC -Wall -Wextra -mrtm -mwaitpkg -std=c++11 -MD -MT src/tbb/CMakeFiles/tbb.dir/exception.cpp.o -MF CMakeFiles/tbb.dir/exception.cpp.o.d -o CMakeFiles/tbb.dir/exception.cpp.o -c /home/x/mold/mold-1.4.0/third-party/tbb/src/tbb/exception.cpp`

**Comment by rui314** [comment]
Let me reproduce the issue on my machine using Docker. What exactly did you run to try to build mesa? Does it happen all the time or with some probability?

**Comment by Nanotwerp** [comment]
> Let me reproduce the issue on my machine using Docker. What exactly did you run to try to build mesa? Does it happen all the time or with some probability?

I modified the spec file of Mesa to use mold by setting CC_LD and CXX_LD to mold. I modified it to use LTO by putting -flto=thin for CFLAGS, CXXFLAGS, and LDFLAGS. Then, I compile using fedpkg. 

The issue happens every time, and it's always at the same part of linking (I don't know how I forgot to include this. Very sorry haha): 

`mold: fatal: cannot open src/gallium/drivers/radeonsi/libradeonsi.a.p/si_get.c.o: Too many open files`

**Comment by rui314** [comment]
What number did you set as the maximum number of open files? Check it out by running `ulimit -a`.

**Comment by Nanotwerp** [comment]
> What number did you set as the maximum number of open files? Check it out by running `ulimit -a`.

Oh God! It was a ulimit issue after all! I tried changing ulimit -n multiple times, but apparently systemd makes changing file limits more complicated, with the graphical login environment not accepting these changed soft limits. Following the steps shown in the answer in [this StackExchange post](https://superuser.com/questions/1200539/cannot-increase-open-file-limit-past-4096-ubuntu) and changing the value of DefaultLimitNOFILE for /etc/systemd/system.conf and /etc/systemd/user.conf to 4096, I got it to work! Thank you so much for your quick and helpful responses!
